pkg/tenant/tenants.go

```diff
@@ -0,0 +1,121 @@
+package tenant
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/rs/zerolog/log"
+)
+
+// CreateTenant creates a new tenant in the database
+func (s *Server) CreateTenant(c *gin.Context) {
+	var (
+		err    error
+		tenant *api.Tenant
+	)
+
+	if err = c.BindJSON(&tenant); err != nil {
+		log.Warn().Err(err).Msg("could not parse tenant create request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse request"))
+		return
+	}
+
+	if tenant.ID == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrTenantIDRequired))
+	}
+
+	if tenant.Name == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant name is required"))
+	}
+
+	if tenant.EnvironmentType == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("environment type is required"))
+	}
+
+	// Add Tenant to the database
+	if err = db.CreateTenant(c.Request.Context(), &db.Tenant{ID: ulid.Make()}); err != nil {
+		log.Error().Err(err).Msg("could not save tenant")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not add tenant"))
+		return
+	}
+
+	c.JSON(http.StatusCreated, tenant)
+}
+
+func (s *Server) TenantList(c *gin.Context) {
+
+	c.JSON(http.StatusNotImplemented, "not implemented yet")
+}
+
+func (s *Server) TenantCreate(c *gin.Context) {
+	var (
+		err    error
+		tenant *api.Tenant
+	)
+
+	// TODO: Add authentication and authorization middleware
+	if err = c.BindJSON(&tenant); err != nil {
+		log.Warn().Err(err).Msg("could not parse tenant create request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
```

Comment:
Don't forget to return after calling `c.JSON`! 

```suggestion
		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
		return
```